pkg/apis/experiment/experiment.go

```diff
@@ -0,0 +1,390 @@
+/*
+Copyright © 2021 The LitmusChaos Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package experiment
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/litmuschaos/litmusctl/pkg/apis"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/litmuschaos/litmus/chaoscenter/graphql/server/graph/model"
+	types "github.com/litmuschaos/litmusctl/pkg/types"
+	"github.com/litmuschaos/litmusctl/pkg/utils"
+)
+
+// CreateExperiment sends GraphQL API request for creating a Experiment
+func CreateExperiment(pid string, requestData model.SaveChaosExperimentRequest, cred types.Credentials) (RunExperimentData, error) {
+
+	// Query to Save the Experiment
+	var gqlReq SaveChaosExperimentGraphQLRequest
+
+	gqlReq.Query = SaveExperimentQuery
+	gqlReq.Variables.ProjectID = pid
+	gqlReq.Variables.SaveChaosExperimentRequest = requestData
+
+	query, err := json.Marshal(gqlReq)
+	if err != nil {
+		return RunExperimentData{}, err
+	}
+
+	resp, err := apis.SendRequest(
+		apis.SendRequestParams{
+			Endpoint: cred.Endpoint + utils.GQLAPIPath,
+			Token:    cred.Token,
+		},
+		query,
+		string(types.Post),
+	)
+	if err != nil {
+		return RunExperimentData{}, err
+	}
+
+	bodyBytes, err := ioutil.ReadAll(resp.Body)
+
+	defer resp.Body.Close()
+	if err != nil {
+		return RunExperimentData{}, errors.New("Error in creating Chaos Experiment: " + err.Error())
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		var savedExperiment SaveExperimentData
+
+		err = json.Unmarshal(bodyBytes, &savedExperiment)
+
+		if err != nil {
+			return RunExperimentData{}, errors.New("Error in creating Chaos Experiment: " + err.Error())
+		}
+
+		// Errors present
+		if len(savedExperiment.Errors) > 0 {
+			return RunExperimentData{}, errors.New(savedExperiment.Errors[0].Message)
+		}
+
+		if strings.Contains(savedExperiment.Data.Message, "experiment saved successfully") {
+			fmt.Print("\n🚀 Chaos Experiment successfully Saved 🎉")
+		}
+	} else {
+		return RunExperimentData{}, errors.New("graphql schema error")
```

Comment:
This need not be a graphql schema error, can you add a generic msg like: error saving the experiment